docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,330 @@
+
+## Naming Conventions
+
+- **Components and pipelines** use `snake_case` (e.g., `data_preprocessing`, `model_trainer`)
+- **Git branches** use `component/name` for features or `fix/issue-123` for bug fixes
+- **Commit messages** follow [Conventional Commits](https://conventionalcommits.org/) format with type prefix (feat, fix, docs, etc.)
+
+### Required Files
+
+Every component must include these files in its directory:
+
+```
+components/<category>/<component_name>/
+├── __init__.py            # Exposes the component function for imports
+├── component.py           # Main implementation
+├── metadata.yaml          # Complete specification (see schema below)
+├── README.md              # Overview, inputs/outputs, usage examples, development instructions
+├── OWNERS                 # Maintainers (at least one Kubeflow SIG owner for Core tier)
+├── Containerfile          # Container definition (required only for Core tier custom images)
+├── example_pipelines.py   # Working usage examples
+└── tests/
+│   └── test_component.py  # Unit tests
+└── <supporting_files>
+```
+
+Similarly, every pipeline must include these files:
+```
+pipelines/<category>/<pipeline_name>/
+├── __init__.py            # Exposes the pipeline function for imports
+├── pipeline.py            # Main implementation
+├── metadata.yaml          # Complete specification (see schema below)
+├── README.md              # Overview, inputs/outputs, usage examples, development instructions
+├── OWNERS                 # Maintainers (at least one Kubeflow SIG owner for Core tier)
+├── example_pipelines.py   # Working usage examples
+└── tests/
+│   └── test_pipeline.py  # Unit tests
+└── <supporting_files>
+```
+
+### metadata.yaml Schema
+
+Your `metadata.yaml` must include these fields:
+
+```yaml
+tier: core  # or 'third_party'
+name: my_component
+stability: stable  # 'alpha', 'beta', or 'stable'
+dependencies:
+  kubeflow:
+    - name: Pipelines
+      version: '>=2.5'
+  external_services:  # Optional list of external dependencies
+    - name: Argo Workflows
+      version: "3.6"
+tags:  # Optional keywords for discoverability
+  - training
+  - evaluation
+lastVerified: 2025-11-18T00:00:00Z  # Updated annually; components are removed after 12 months without update
+ci:
+  compile_check: true  # Validates component compiles with kfp.compiler
+  skip_dependency_probe: false  # Set true only with justification
+  pytest: optional  # Set to 'required' for Core tier
+packages_to_install:  # Dependencies verified by CI
+  - pandas>=1.5.0
+  - scikit-learn>=1.2.0
+links:
+  documentation: https://kubeflow.org/components/my_component
+  issue_tracker: https://github.com/kubeflow/pipelines-components/issues
+```
+
+### OWNERS File
+
+The OWNERS file enables component owners to self-service maintenance tasks including approvals, metadata updates, and lifecycle management:
```

Comment:
can you either link to the [k8s docs](https://docs.prow.k8s.io/docs/components/plugins/approve/approvers/#lgtm-label) or add a brief description on `/lgtm `and `/approve` semantics

as I believe we intend to use prow like the rest of the kf projects